app/controllers/api/v1/votes_controller.rb

```diff
@@ -0,0 +1,54 @@
+class Api::V1::VotesController < ApiController
+	before_action :authenticate_user
+
+  def create
+		existing_vote = Vote.where(user_id: current_user.id, review_id: params['review_id']).first
+
+    if existing_vote
+      review = Review.find(params['review_id'])
+
+      if existing_vote.vote_type === params['vote_type']
+        existing_vote.destroy
+
+        karma = review.total_karma
+        vote = { vote_type: nil }
+        payload = { vote: vote, karma: karma }
+
+        render json: payload
+      else
+        existing_vote.update(vote_type: params['vote_type'])
+
+        karma = review.total_karma
+        payload = { vote: existing_vote, karma: karma}
+
+        render json: payload
+      end
+
+    else
+      vote_hash = vote_params.merge!({user_id: current_user.id})
+      vote = Vote.new(vote_hash)
+
+      if vote.save
+        review = Review.find(vote.review_id)
+        karma = review.total_karma
+        payload = { vote: vote, karma: karma }
+
+        render json: payload
+      else
+        render json: {error: "An error occurred."}
+      end
+    end
+  end
```

Comment:
aside from the indentation (which needs to be corrected) very impressed. Its just a really long method. This is optional, but I would encourage potentially making a methods at the bottom of the controller to reduce the amount of logic in this action. that way it will read better and look simpler

Reply:
Updated the create method to have two separate methods "change_existing_vote" and "create_new_vote" which are called depending on whether an existing_vote is found or nil.